src/utils/transaction.ts

```diff
@@ -138,3 +186,33 @@ export async function getSyncTransactions(
 
   return syncTransactionsWithState(newTransactions, stateTransactions);
 }
+
+const tokenRegex = /Left[(]Left[(]Left[(]Pair"([A-Za-z0-9]*)"[(]Pair"([A-Za-z0-9]*)["]([0-9]*)[))))]/;
```

Comment:
This is not the best place to put it, different tokens will have different shape transactions. This should be in the token component.